core/src/main/java/com/sequenceiq/cloudbreak/reactor/handler/cluster/StopStartDownscaleDecommissionViaCMHandler.java

```diff
@@ -91,16 +97,36 @@ protected Selectable doAccept(HandlerEvent<StopStartDownscaleDecommissionViaCMRe
             HostGroup hostGroup = hostGroupService.getByClusterIdAndName(cluster.getId(), request.getHostGroupName())
                     .orElseThrow(NotFoundException.notFound("hostgroup", request.getHostGroupName()));
 
+            List<InstanceMetaData> instancesWithServicesNotRunning = cloudInstanceIdToInstanceMetaDataConverter.getNotDeletedAndNotZombieInstances(stack,
+                    request.getHostGroupName(),
+                    request.getRunningInstancesWithServicesNotRunning().stream()
+                            .map(CloudInstance::getInstanceId)
+                            .collect(Collectors.toSet()));
+
             Map<String, InstanceMetaData> hostsToRemove = clusterDecomissionService.collectHostsToRemove(hostGroup, hostNames);
             List<String> missingHostsInCm = Collections.emptyList();
+            Map<String, InstanceMetaData> finalHostsToRemove = hostsToRemove;
             if (hostNames.size() != hostsToRemove.size()) {
                 missingHostsInCm = hostNames.stream()
-                        .filter(h -> !hostsToRemove.containsKey(h))
+                        .filter(h -> !finalHostsToRemove.containsKey(h))
                         .collect(Collectors.toList());
                 LOGGER.info("Found fewer instances in CM to decommission, as compared to initial ask. foundCount={}, initialCount={}, missingHostsInCm={}",
                         hostsToRemove.size(), hostNames.size(), missingHostsInCm);
             }
 
+            List<InstanceMetaData> additionalInstancesWithServicesNotRunningToDecommission = instancesWithServicesNotRunning
+                    .stream().filter(i -> !finalHostsToRemove.containsKey(i.getDiscoveryFQDN())).collect(Collectors.toList());
+
+            if (!additionalInstancesWithServicesNotRunningToDecommission.isEmpty()) {
+                LOGGER.info("Including running instances with services not running as part of hostsToDecommission");
```

Comment:
Are we differentiating between nodes where services are not running (i.e. the decommission succeeded, but the STOP failed). At the moment all such nodes would be part of this list, which means this will invoke a decommission on a host which is already decommissioned. How does CM behave in this scenario?
Similarly for nodes where maintenance mode may have been skipped. We should probably be putting nodes into maintenance before decommissioning them.